### DISCUSSION TITLE
model i3-desktop AND Xeon server

### DISCUSSION THREAD

**Discussion by stiebel** [ltsp/ltsp#892] [General]
Hi,
I want to use LTSP with a farm of 100 diskless desktop computer and a special server.
I want to use USB for arduino programming, so the users have to be in the "dialout" group and need access to the ports.
I do not want to have these user accounts in the dialout group of the server.
So I cannot (or do not want to) use the chrootless option on this very server.

What I would like to do is using the chrotless version on one of the clients as a default model but then transfer the image to the server and use it from this bigger machine. It is ok if I have to switch of the model client after that as I have a spare dektop computer just for this purpose.

My questions are:
1) has someone done this before and can give me a hand?
2) is it straight forward and easy enough that I do not need 1)
3) which files to I have to copy, which files do I have to alter?
4) Is there anything else to take care of?

Thanks
Roland

**Comment by alkisg** [comment]
An easy way to give users access to arduino without involving any groups is to add an udev rule like the following:

```
# Original Arduino (/dev/ttyUSB0): Arduino SA Uno R3 (CDC ACM)
ATTRS{idVendor}=="2341", ATTRS{idProduct}=="0043", SUBSYSTEMS=="usb", ACTION=="add", MODE="0666"
```

Another way would be: https://enotty.pipebreaker.pl/2012/05/23/linux-automatic-user-acl-management/

The easiest way is chrootless,
the second easiest is in a VM (e.g. VirtualBox),
and of course it's also possible to prepare a client the way you want and then transfer it to the server e.g. via ssh and dd in the form of an image or chroot; I documented a similar case in https://ltsp.org/docs/installation/raspios/#chroot-preparation. But note that you won't be transferring /home/username; you'll only be transferring /. That means you won't transfer user settings and programs; but only system settings and programs.

All methods are pretty much equivalent regarding safety; you can limit your users to only sftp access in all these cases.

> has someone done this before and can give me a hand?

If you're asking about professional support, sure, feel free to email me at alkisg gmail; otherwise github discussions are fine, as time permits.
